### PR TITLE
Split Frontend and DCR fixtures

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

Split Frontend and DCR fixtures

## Why?

As part of the [`gen-fixtures`](https://github.com/guardian/dotcom-rendering/blob/4c2a5055f4ae52a0da036b3a6362e47687779a93/dotcom-rendering/scripts/test-data/gen-fixtures.js) script we currently output articles using the enhanced [DCRArticle](https://github.com/guardian/dotcom-rendering/blob/23c1750e259038b2240c2ea37b7373d247936d58/dotcom-rendering/src/types/frontend.ts#L139) type. 

Playwright tests use the DCRArticle fixtures but should strictly use the [FEArticle](https://github.com/guardian/dotcom-rendering/blob/23c1750e259038b2240c2ea37b7373d247936d58/dotcom-rendering/src/types/frontend.ts#L20) type as it will POST the JSON to DCR mimicking a POST from Frontend.

This currently works as the DCRArticle type extends the FEArticle type but from https://github.com/guardian/dotcom-rendering/pull/10801 the types will diverge so this changes lays the groundwork for the types to be split.

There's quite a bit of noise from the generated fixtures so I've added comments highlighting the main changes.

### CODE REVIEW

File: dotcom-rendering/scripts/test-data/gen-fixtures.js [hunk 1]
```diff
@@ -146,48 +147,68 @@ const HEADER = `/**
 const requests = articles.map((article) => {
 	return fetch(`${article.url}.json?dcr`)
 		.then((res) => res.json())
-		.then((json) => {
+		.then((frontendJson) => {
```
Comment (arelra) [inline dotcom-rendering/scripts/test-data/gen-fixtures.js:150]: Main changes to `gen-fixtures` are here

File: dotcom-rendering/playwright/lib/load-page.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import type { Page } from '@playwright/test';
 import { validateAsArticleType } from '../../src/model/validate';
-import type { DCRArticle, FEArticleType } from '../../src/types/frontend';
+import type { FEArticleType } from '../../src/types/frontend';
```
Comment (arelra) [inline dotcom-rendering/playwright/lib/load-page.ts:3]: Playwright's `load-page` now only uses FEArticleType

File: dotcom-rendering/scripts/test-data/gen-fixtures.js [hunk 3]
```diff
@@ -146,48 +147,68 @@ const HEADER = `/**
...
-			export const ${article.name}: DCRArticle = ${JSON.stringify(
-				enhancedArticle,
-				null,
-				4,
-			)}`;
-			return fs.writeFile(
-				`${root}/fixtures/generated/articles/${article.name}.ts`,
-				contents,
+			const dcrTypeFile = fs.writeFile(
+				`${root}/fixtures/generated/dcr-articles/${article.name}.ts`,
```
Comment (arelra) [inline dotcom-rendering/scripts/test-data/gen-fixtures.js:189]: dcrArticle types are written to `${root}/fixtures/generated/dcr-articles`

File: dotcom-rendering/scripts/test-data/gen-fixtures.js [hunk 4]
```diff
@@ -146,48 +147,68 @@ const HEADER = `/**
...
+
+				export const ${article.name}: FEArticleType = ${JSON.stringify(
+					frontendJson,
+					null,
+					4,
+				)}
+			`;
+
+			const frontendTypeFile = fs.writeFile(
+				`${root}/fixtures/generated/fe-articles/${article.name}.ts`,
```
Comment (arelra) [inline dotcom-rendering/scripts/test-data/gen-fixtures.js:206]: frontend article types are written to `${root}/fixtures/generated/fe-articles`

File: dotcom-rendering/fixtures/config.js [hunk 5]
```diff
@@ -94,6 +94,7 @@ module.exports = {
 			inizio: true,
 			prebidHeaderBidding: true,
 			a9HeaderBidding: true,
+			lightbox: true,
```
Comment (arelra) [inline dotcom-rendering/fixtures/config.js:97]: Set lightbox switch to true for all fixtures as it wasn't included before

**Comment by arelra** [comment]
Not a regression but a new chromatic diff on the `DecideLayout:Web Liveblog With No Key Events` snapshot:

https://www.chromatic.com/test?appId=63e251470cfbe61776b0ef19&id=65fd438ed9ef596ae5989096



The diff is caused by addition of the 'Ukraine' menu item to the subnav. This extra item causes the nav to extend and activating [collapsed styling](https://github.com/guardian/dotcom-rendering/blob/216d54c56c1b2adf154e36bb0ea610489724551c/dotcom-rendering/src/components/SubNav.importable.tsx#L172). This sets a max-width on the subnav and with the snapshot width of 1200px this causes the the last two items to be disappear from the nav.

You can see this happening in this video:

https://github.com/guardian/dotcom-rendering/assets/7014230/b62ea13d-8c18-4cbd-90b8-ef54b5c47bcf

As this is expected behaviour I'll accept the diff

**Comment by arelra** [comment]
Regenerating the fixtures in PR brought to light a couple of regressions from Frontend:

- generated tag links, fixed here: https://github.com/guardian/frontend/pull/26984
- subnav missing from articles with the tag `us-news/us-politics` caused by a side effect of this change: https://github.com/guardian/frontend/pull/26840. Raised a bug here: https://github.com/guardian/dotcom-rendering/issues/11005

So we don't hold up this PR I'll accept the diffs and when/if we fix the subnav bug we can regenerate the fixtures and snapshots again.

File: dotcom-rendering/src/model/enhance-dividers.test.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { Standard as ExampleArticle } from '../../fixtures/generated/articles/Standard';
+import { Standard as ExampleArticle } from '../../fixtures/generated/dcr-articles/Standard';
```
Comment (JamieB-gu) [inline dotcom-rendering/src/model/enhance-dividers.test.ts:1]: Would it be correct to say the enhancers essentially do `FEArticleType -> DCRArticle`? If so, does that mean these enhancer test files should be importing `fe-articles`?

File: dotcom-rendering/src/server/handler.article.amp.tsx [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import { ArticleDesign } from '@guardian/libs';
 import type { RequestHandler } from 'express';
-import { Standard as ExampleArticle } from '../../fixtures/generated/articles/Standard';
+import { Standard as ExampleArticle } from '../../fixtures/generated/dcr-articles/Standard';
```
Comment (JamieB-gu) [inline dotcom-rendering/src/server/handler.article.amp.tsx:3]: If this is used to mock a frontend request for the performance test handler should it be using `fe-articles`?

```suggestion
import { Standard as ExampleArticle } from '../../fixtures/generated/fe-articles/Standard';
```

File: dotcom-rendering/src/server/handler.article.web.ts [hunk 8]
```diff
@@ -1,5 +1,5 @@
 import type { RequestHandler } from 'express';
-import { Standard as ExampleArticle } from '../../fixtures/generated/articles/Standard';
+import { Standard as ExampleArticle } from '../../fixtures/generated/dcr-articles/Standard';
```
Comment (JamieB-gu) [inline dotcom-rendering/src/server/handler.article.web.ts:2]: If this is used to mock a frontend request for the performance test handler should it be using `fe-articles`?

```suggestion
import { Standard as ExampleArticle } from '../../fixtures/generated/fe-articles/Standard';
```

**Review by JamieB-gu — COMMENTED** [review]
Looks good! Couple of questions/suggestions about using `fe-articles` in more places.

File: dotcom-rendering/src/server/handler.article.amp.tsx (hunk 7 above)
Comment (arelra) [reply dotcom-rendering/src/server/handler.article.amp.tsx]: Good spot thanks. Have updated all the affected server handlers

**Review by JamieB-gu — APPROVED** [review]

**Pull request merged**